### DISCUSSION TITLE
ActiveJ http and MDC/Thread local

### DISCUSSION THREAD

**Discussion by Evgen1000end** [activej/activej#265] [Q&A]
In my typical web application (Spring MCV) i actively used MDC context and thread local variables to store traceId in them for use in logs, 
for distributed traces, etc.

Now similar logic needs to be implemented in ActiveJ, but it seems that you can’t use thread local variables here (in fact, I have a single-threaded application with one event-loop).

What is the correct way to solve this problem?
Throw traceId through the context, through all the promises explicitly? Maybe there is some more generic way?

Thanks in advance!

**Comment by eduard-vasinskyi** [comment]
Hi, @Evgen1000end

I have answered a similar question [here](https://github.com/activej/activej/discussions/264#discussioncomment-4799827). Hope that answers your question.
